### PR TITLE
Port `su` program

### PR DESCRIPTION
**Pull request by franciscozdo**
Depends on changes in #766 and #767.

### CODE REVIEW

File: etc/master.passwd [hunk 1]
```diff
@@ -1,2 +1,2 @@
 root::0:0::0:0:root:/root:/bin/ksh
-
+paco::1000:1000::0:0:paco:/root:/bin/ksh
```
Comment (pj1031999) [inline etc/master.passwd:2]: Is _paco_ a special username? Why not something like _user_ or _mimiker_?
I'm wonder if `/root` is a good home directory for any user other than _root_.

File: lib/libc/gen/initgroups.c [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+#include <errno.h>
+#include <grp.h>
+
+int initgroups(const char *uname, gid_t agroup) {
+  gid_t groups_list[NGROUPS];
+  int ngroups;
+  gid_t *groups = groups_list;
+  int rval;
+
+  //_DIAGASSERT(uname != NULL);
```
Comment (pj1031999) [inline lib/libc/gen/initgroups.c:48]: Why this line is commented? Do you plan to use it in the future?

File: usr.bin/su/Makefile [hunk 3]
```diff
@@ -0,0 +1,11 @@
+TOPDIR = $(realpath ../..)
+
+PROGRAM = su
+SOURCES = su.c suutil.c
+FORMAT-EXCLUDE = $(SOURCES)
+
+LDLIBS = -lutil
+
+#CPPFLAGS += -DALLOW_GROUP_CHANGE -DALLOW_EMPTY_USER
```
Comment (pj1031999) [inline usr.bin/su/Makefile:9]: Why this line is commented? Do you plan to use these flags later?

File: usr.bin/su/suutil.h [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+ * HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY
+ * OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF
+ * SUCH DAMAGE.
+ */
+
+__BEGIN_DECLS
+int chshell(const char *);
+char *ontty(void);
+__END_DECLS
```
Comment (pj1031999) [inline usr.bin/su/suutil.h:35]: `__BEGIN_DECLS` and `__END_DECLS` are defined in `sys/cdefs.h`. How can we see them here?

**Comment by pj1031999** [comment]
I see a lot of `#if 0 ... #endif` in code. Is it ok?

File: usr.bin/su/suutil.h (hunk 4 above)
Comment (franciscozdo) [reply usr.bin/su/suutil.h:35]: done.

File: include/sys/param.h [hunk 5]
```diff
@@ -111,6 +111,8 @@
 
 #define powerof2(x) ((((x)-1) & (x)) == 0)
 
+#define NGROUPS NGROUPS_MAX
```
Comment (cahirwpz) [inline include/sys/param.h:114]: This is already defined but earlier.

File: lib/libc/gen/initgroups.c (hunk 2 above)
Comment (franciscozdo) [reply lib/libc/gen/initgroups.c]: removed

File: usr.bin/su/Makefile (hunk 3 above)
Comment (franciscozdo) [reply usr.bin/su/Makefile]: Yes. Flag `DALLOW_GROUP_CHANGE` could be used in the future. `DALLOW_EMPTY_USER` is now used.

File: include/sys/param.h (hunk 5 above)
Comment (franciscozdo) [reply include/sys/param.h]: Removed.

File: etc/master.passwd (hunk 1 above)
Comment (franciscozdo) [reply etc/master.passwd]: For now changed to users 'admin' and 'user'. Home directory for all users is now /

**Pull request closed without merging**